utils/generator/generate-api.py

```diff
@@ -313,9 +416,14 @@ def from_schema(cls, filepath: pathlib.Path):
         with filepath.open() as f:
             schema_data = json.loads(f.read())
 
-        def expand_refs(x):
+        def expand_refs(x, depth=0):
+            # Hack to prevent recursive expansion, we only
+            # really need top-most layers expanded.
+            if depth > 10:
```

Comment:
There's some reference loops within `requestBody` schemas that needs resolving, for now this is an easy way to ensure we resolve all `$ref` instances as far as we need without writing a "better" ref resolver.

Reply:
Do you want to prevent circular reference in `$ref`? I found [this comment](https://github.com/OAI/OpenAPI-Specification/issues/822#issuecomment-624424053) about circular ref in OpenAPI.